### PR TITLE
Refactor OpenerService. Fix #79487

### LINKED ISSUE #79487: Opener Service: trusted domains, standalone editor, openExternal
Adding @alexandrudima for standalone editor.

@octref it looks like you added support for showing a dialog to the user whenever we open a URL externally into `IOpenerService`. I think this makes sense for the case of the workbench, but the way it works now, it will also trigger a dialog in the standalone editor (I am not even sure we have support for dialogs there). To make matters worse, we store the result into the storage service, which as far as I know is just implemented in-memory in the standalone editor case. So anytime a link opens, the user would get asked again after a reload of the page.

Also, I see some layering issues in the code: You seem to reference `workbench.action.configureTrustedDomains` from `vs/editor` which is not good. 

Here is my suggestion:
* remove this logic from the opener service in `vs/editor/browser/services`
* move this logic, including the code that contributes the `workbench.action.configureTrustedDomains` into a single workbench contribution (`url.contribution.ts` for example)
* leverage the `IOpenerService.registerOpener()` method from the workbench layer to participate in the opening and do the dialog + storage logic from there

On top of that: I recently added `IOpenerService#openExternal()` which is the code that gets triggered, e.g. when an extension calls `vscode.env.openExternal()`. I wonder: should we show the same dialog on that case for trusted domains? I would think so, but keep in mind that one of the URLs may have the `vscode://` scheme which we should always trust.

### PR DESCRIPTION
**Pull request by octref**

### CODE REVIEW

File: src/vs/editor/browser/services/openerService.ts [hunk 1]
```diff
@@ -4,49 +4,61 @@
...
 import { ServiceIdentifier } from 'vs/platform/instantiation/common/instantiation';
+import { IOpener, IOpenerService, IValidator } from 'vs/platform/opener/common/opener';
 
-export class OpenerService implements IOpenerService {
+export class OpenerService extends Disposable implements IOpenerService {
 
 	_serviceBrand!: ServiceIdentifier<any>;
 
 	private readonly _opener = new LinkedList<IOpener>();
+	private readonly _validatorMap: { [k: string]: LinkedList<IValidator> } = {};
```
Comment (bpasero) [inline src/vs/editor/browser/services/openerService.ts]: Use `Map` here?

File: src/vs/editor/browser/services/openerService.ts [hunk 2]
```diff
@@ -4,49 +4,61 @@
...
-		//
+		super();
 	}
 
 	registerOpener(opener: IOpener): IDisposable {
 		const remove = this._opener.push(opener);
 		return { dispose: remove };
 	}
 
+	registerValidator(uriScheme: string, validator: IValidator): IDisposable {
```
Comment (bpasero) [inline src/vs/editor/browser/services/openerService.ts]: Why not just for all URLs, why scoped to scheme? We do not do that for the opener right?

File: src/vs/editor/browser/services/openerService.ts [hunk 3]
```diff
@@ -158,22 +130,10 @@ export class OpenerService implements IOpenerService {
...
- */
-function isDomainTrusted(domain: string, trustedDomains: string[]) {
-	for (let i = 0; i < trustedDomains.length; i++) {
-		if (trustedDomains[i] === '*') {
-			return true;
-		}
-
-		if (trustedDomains[i] === domain) {
-			return true;
+	dispose() {
```
Comment (bpasero) [inline src/vs/editor/browser/services/openerService.ts]: I do not think this is necessary, but if you go here, then also clear the opener Set. If you use a `Map` you can call `map.clear();`

**Review by bpasero — CHANGES_REQUESTED** [review]
Direction is good, some feedback provided.

**Comment by octref** [comment]
OK, should be good to go.

**Review by bpasero — APPROVED** [review]
@octref thanks, feel free to merge. Btw I recommend to use async/await over `then` :)

**Pull request merged**